### ISSUE TITLE
Hyprland lock allows inputs to pass through

### ISSUE THREAD

**Issue by acheong08** [hyprwm/Hyprland#5897]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit c7fbc30bfd64b2104006fef291f4c4bff537beb2 dirty (Nix: add missing deps).
Date: Sun May 5 18:34:14 2024
Tag: v0.40.0-10-gc7fbc30b, commits: 4616

flags: (if any)


System Information:
System name: Linux
Node name: fish
Release: 6.8.8-300.fc40.x86_64
Version: #1 SMP PREEMPT_DYNAMIC Sat Apr 27 17:53:31 UTC 2024


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation TigerLake-LP GT2 [Iris Xe Graphics] [8086:9a49] (rev 01) (prog-if 00 [VGA controller])


os-release: NAME="Fedora Linux"
VERSION="40 (Workstation Edition)"
ID=fedora
VERSION_ID=40
VERSION_CODENAME=""
PLATFORM_ID="platform:f40"
PRETTY_NAME="Fedora Linux 40 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:40"
DEFAULT_HOSTNAME="fedora"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f40/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=40
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=40
SUPPORT_END=2025-05-13
VARIANT="Workstation Edition"
VARIANT_ID=workstation


plugins:


```

</details>

### Bug or Regression?

Bug

### Description

Reference from Hyprlock: https://github.com/hyprwm/hyprlock/issues/305

### How to reproduce

```
$ swaylock
```
Press a keyboard shortcut you have enabled. In my case, I use `SUPER+Q` to open a terminal

Unlock

You'll see a window open.

### Crash reports, logs, images, videos

```
[LOG] Callback 4122de8 -> 4122de0, FractionalScale removed.
[LOG] Searching for matching rules for foot (title: clear)
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 40eebe0: title: "clear"]
[LOG] Window 40eebe0 set title to clear
[LOG] Searching for matching rules for foot (title: acheong@fish:/var/run/user/1000/hypr)
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 40eebe0: title: "acheong@fish:/var/run/user/1000/hypr"]
[LOG] Window 40eebe0 set title to acheong@fish:/var/run/user/1000/hypr
[LOG] Searching for matching rules for foot (title: tail -f )
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 40eebe0: title: "tail -f "]
[LOG] Window 40eebe0 set title to tail -f 
[LOG] Keybind triggered, calling dispatcher (64, , 108)
[LOG] Executing swaylock -i $(swww query | awk '{ print $8 }')
[LOG] Process Created with pid 91460
[LOG] [SessionLock] New sessionLock with id 3
[LOG] Session got locked by 4128960
[LOG] [SessionLock] New sessionLockSurface with id 16
[LOG] Registered signal for owner 35182e0: 41518d8 -> 3518398 (owner: SessionLockSurface)
[LOG] Registered signal for owner 35182e0: 4151918 -> 3518400 (owner: SessionLockSurface)
[LOG] Registered signal for owner 4151600: 4151918 -> 411e090 (owner: FractionalScale)
[LOG] Set keyboard focus to surface 4151600
[LOG] Callback 34d5738 -> 34d5730, CTextInput removed.
[LOG] Callback 34d57a0 -> 34d5798, CTextInput removed.
[WARN] Disabled TextInput commit?
[ERR] BUG THIS: key not found in m_dPressedKeys
[LOG] Keybind triggered, calling dispatcher (64, , 102)
[LOG] Executing librewolf --ProfileManager
[LOG] Process Created with pid 91469
[LOG] [TextInputV3] New tiv3 at 00000000041347b0
[LOG] New XDG Toplevel created. (class: null)
[LOG] Registered signal for owner 41b5f00: 41e9498 -> 41b5f68 (owner: XDG Window)
[LOG] Registered signal for owner 41b5f00: 412d3a8 -> 41b6038 (owner: XDG Window)
[LOG] Registered signal for owner 41b5f00: 41e9488 -> 41b5f00 (owner: XDG Window)
[LOG] Registered signal for owner 41b6720: 41e94c8 -> 41b67a0 (owner: CWLSurface)
[LOG] Registered signal for owner 41b6720: 41e9488 -> 41b6808 (owner: CWLSurface)
[LOG] CWLSurface 41b6720 called init()
[LOG] Searching for matching rules for librewolf-default (title: LibreWolf - Choose User Profile)
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 41b5f00: title: ""]
[LOG] Layout predicts size [Vector2D: x: 564, y: 716] for [Window 41b5f00: title: ""]
[LOG] New window contains HL_INITIAL_WORKSPACE_TOKEN: f47ed947-bb20-4e75-a319-ee8be1ca8315
[LOG] HL_INITIAL_WORKSPACE_TOKEN f47ed947-bb20-4e75-a319-ee8be1ca8315 -> 3
[LOG] Registered signal for owner 41b5f00: 41e94a8 -> 41b5fd0 (owner: CWindow)
[LOG] Registered signal for owner 413bd80: 41e94b8 -> 413be50 (owner: CSubsurface Head)
[LOG] Registered signal for owner 41395b0: 41b7668 -> 4139630 (owner: CWLSurface)
[LOG] Registered signal for owner 41395b0: 41b7628 -> 4139698 (owner: CWLSurface)
[LOG] CWLSurface 41395b0 called init()
[LOG] Registered signal for owner 41393a0: 41b7628 -> 4139408 (owner: CSubsurface)
[LOG] Registered signal for owner 41393a0: 40f06d0 -> 41393a0 (owner: CSubsurface)
[LOG] Registered signal for owner 41393a0: 41b7658 -> 4139470 (owner: CSubsurface)
[LOG] Registered signal for owner 41393a0: 41b7638 -> 41394d8 (owner: CSubsurface)
[LOG] Registered signal for owner 41393a0: 41b7648 -> 4139540 (owner: CSubsurface)
[LOG] Registered signal for owner 41b8fe0: 412d3c8 -> 41b9198 (owner: CPopup Head)
[LOG] Searching for matching rules for librewolf-default (title: LibreWolf - Choose User Profile)
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 41b5f00: title: "LibreWolf - Choose User Profile"]
[LOG] OPENINGON: [Node 419f080: workspace: 3, pos: [0.00, 36.00], size: [1128.00, 716.00], window: 40eebe0], Monitor: 0
[LOG] Searching for matching rules for foot (title: tail -f )
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 40eebe0: title: "tail -f "]
[LOG] Searching for matching rules for librewolf-default (title: LibreWolf - Choose User Profile)
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 41b5f00: title: "LibreWolf - Choose User Profile"]
[LOG] Registered signal for owner 41b5f00: 412c568 -> 41b60a0 (owner: XDG Window Late)
[LOG] Registered signal for owner 41b5f00: 412c4f8 -> 41b62a8 (owner: XDG Window Late)
[LOG] Registered signal for owner 41b5f00: 412c518 -> 41b6240 (owner: XDG Window Late)
[LOG] Registered signal for owner 41b5f00: 412c528 -> 41b61d8 (owner: XDG Window Late)
[LOG] Registered signal for owner 41b5f00: 412c538 -> 41b6310 (owner: XDG Window Late)
[LOG] Registered signal for owner 41b5f00: 412c508 -> 41b6170 (owner: XDG Window Late)
[LOG] Registered signal for owner 41b5f00: 412d3e8 -> 41b66b8 (owner: XDG Window Late)
[LOG] Registered signal for owner 41e91b0: 41e94c8 -> 41b7ac0 (owner: FractionalScale)
[LOG] Map request dispatched, monitor eDP-1, window pos: [565.00000, 36.00000], window size: [563.00000, 716.00000]
[LOG] [ForeignToplevelWlr] Newly mapped window 00000000041b5f00
[LOG] Searching for matching rules for foot (title: tail -f )
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 40eebe0: title: "tail -f "]
[LOG] Searching for matching rules for librewolf-default (title: LibreWolf - Choose User Profile)
[LOG] Window rule suppressevent maximize ->  class:.* matched [Window 41b5f00: title: "LibreWolf - Choose User Profile"]
[LOG] cursorImage request: surface 41c04c0
[LOG] Registered signal for owner 28b79b8: 41c07d8 -> 28b7a38 (owner: CWLSurface)
[LOG] Registered signal for owner 28b79b8: 41c0798 -> 28b7aa0 (owner: CWLSurface)
[LOG] CWLSurface 28b79b8 called init()
[LOG] cursorImage request: surface 41c04c0
[LOG] cursorImage request: surface 41c04c0
[LOG] Registered signal for owner 41b7350: 41b7668 -> 3fd5670 (owner: FractionalScale)
[WARN] [SessionLock] SessionLockSurface object remains but surface is being destroyed???
[LOG] Callback 35183c0 -> 35183b8, SessionLockSurface removed.
[LOG] Callback 3518428 -> 3518420, SessionLockSurface removed.
[LOG] Callback 411e0b8 -> 411e0b0, FractionalScale removed.
```

I have run `git bisect` and here are the results:
```
90e1411315065ead5b5a7873be4109bdfb9f5470 is the first bad commit
commit 90e1411315065ead5b5a7873be4109bdfb9f5470
Author: Vaxry <vaxry@vaxry.net>
Date:   Tue Apr 30 16:32:05 2024 +0100

    session-lock: move to new impl
```
It may be the few commits after that as this commit also causes a red screen. 
See:
- https://github.com/hyprwm/Hyprland/commit/dbb8b294d64c7a96692bf5c2485441223ef82d2e
- https://github.com/hyprwm/Hyprland/commit/a09103cd38b934a03efdf86c785d08ed6d127bff

**Comment by Agent00Ming** [comment]
fixed by cddeec47a1fc0e70d8598fd10c29cd0e9489999f ?

**Comment by acheong08** [comment]
vaxry is crazy fast
